src/puppet.spec.ts

```diff
@@ -69,6 +69,9 @@ class PuppetTest extends Puppet {
   public async contactList ()                    : Promise<string[]> { return {} as any }
   public async contactQrcode (contactId: string) : Promise<string> { return { contactId } as any }
 
+  public async contactSelfName (newName: string) : Promise<boolean> { return true }
+  public async contactSelfSignature (signature: string) : Promise<boolean> { return true }
```

Comment:
We should follow the current return type like we always used `Promise<void>` instead of others.

`boolean` does not make sense because if it fails, there must be an Error, and it should be thrown out instead of return a `false`.

Reply:
Will change this to `Promise<void>`

Reply:
Ok, thanks.

BTW: does that mean you agree with me about this design: it should throw an exception instead of return false?

Reply:
Yes, I agree with the design for updating actions: no return value, when fail, exception is expected.